instance/tests/models/test_instance.py

```diff
@@ -555,6 +558,50 @@ def test_provision_mongo_again(self):
         self.check_mongo(instance)
 
 
+@patch('instance.openstack.SwiftConnection')
+class SwiftContainerInstanceTestCase(TestCase):
+    """
+    Tests for Swift container provisioning.
+    """
+    def check_swift(self, instance, mock_swift_connection):
+        """
+        Verify Swift settings on the instance and the number of calls to the Swift API.
+        """
+        self.assertIs(instance.swift_provisioned, True)
+        self.assertEqual(instance.swift_openstack_user, settings.SWIFT_OPENSTACK_USER)
+        self.assertEqual(instance.swift_openstack_password, settings.SWIFT_OPENSTACK_PASSWORD)
+        self.assertEqual(instance.swift_openstack_tenant, settings.SWIFT_OPENSTACK_TENANT)
+        self.assertEqual(instance.swift_openstack_auth_url, settings.SWIFT_OPENSTACK_AUTH_URL)
+        self.assertEqual(instance.swift_openstack_region, settings.SWIFT_OPENSTACK_REGION)
+        self.assertCountEqual(
+            [call(c, headers={'X-Container-Read': '.r:*'}) for c in instance.swift_container_names],
+            mock_swift_connection.return_value.put_container.call_args_list,
+        )
+
+    def test_provision_swift(self, mock_swift_connection):
+        """
+        Test provisioning Swift containers, and that they are provisioned only once.
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        instance.provision_swift()
+        self.check_swift(instance, mock_swift_connection)
+
+        # Provision again without resetting the mock.  The assertCountEqual assertion will verify
+        # that the container isn't provisioned again.
+        instance.provision_swift()
+        self.check_swift(instance, mock_swift_connection)
+
+    @override_settings(SWIFT_ENABLE=False)
+    def test_swift_disabled(self, mock_swift_connection):
+        """
+        Verify disabling Swift provisioning works.
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        instance.provision_swift()
+        self.assertIs(instance.swift_provisioned, False)
+        self.assertFalse(mock_swift_connection.called)
+
+
```

Comment:
We should also add some tests to `OpenEdXInstanceTestCase`, to test that `provision_swift` is/is not invoked, and that swift ansible vars are/are not set with different combinations of values for `SWIFT_ENABLE` and `use_ephemeral_databases`.